src/bin/cargo-fmt.rs

```diff
@@ -139,28 +154,113 @@ pub struct Target {
     kind: TargetKind,
 }
 
+#[derive(Debug)]
+pub enum WorkspaceHitlist {
+    All,
+    Some(Vec<String>),
+    None,
+}
+
+impl WorkspaceHitlist {
+    fn is_all(&self) -> bool {
+        match *self {
+            WorkspaceHitlist::All => true,
+            _ => false,
+        }
+    }
+
+    fn is_none(&self) -> bool {
+        match *self {
+            WorkspaceHitlist::None => true,
+            _ => false,
+        }
+    }
+
+    pub fn get_some<'a>(&'a self) -> Option<&'a [String]> {
+        use std::borrow::Borrow;
+        match self {
+            &WorkspaceHitlist::Some(ref hitlist) => Some(hitlist.borrow()),
```

Comment:
You should be able to use `&hitlist` (or possibly `&*hitlist` if that doesn't work) rather than `borrow`.